packages/components/src/Autocomplete/Menu.tsx

```diff
@@ -130,50 +101,41 @@ export function Menu({
           Math.min(options.length - 1, highlightedIndex + indexChange),
         );
       }
-      if (menuDiv.current && menuItemRefs.current[highlightedIndex]) {
-        scrollDownIfTooLow(
-          menuDiv.current,
-          menuItemRefs.current[highlightedIndex],
-        );
+      if (menuDiv.current) {
+        scrollMenuIfItemNotInView(menuDiv.current, "down");
       }
     });
 
-    function scrollDownIfTooLow(
-      menuDivElement: HTMLDivElement,
-      itemDivElement: HTMLDivElement,
-    ) {
-      const tempDiv = itemDivElement.parentElement as HTMLDivElement;
-      const itemHeight = tempDiv.getBoundingClientRect().height;
-      const itemTrueBottom =
-        tempDiv.getBoundingClientRect().bottom + itemHeight;
-      if (itemTrueBottom > menuDivElement.getBoundingClientRect().bottom) {
-        menuDivElement.scrollTop += itemHeight;
-      }
-    }
-
     useOnKeyDown("ArrowUp", (event: KeyboardEvent) => {
       const indexChange = arrowKeyPress(event, IndexChange.Previous);
       if (indexChange) {
         setHighlightedIndex(Math.max(0, highlightedIndex + indexChange));
       }
-      if (menuDiv.current && menuItemRefs.current[highlightedIndex]) {
-        scrollUpIfTooHigh(
-          menuDiv.current,
-          menuItemRefs.current[highlightedIndex],
-        );
+      if (menuDiv.current) {
+        scrollMenuIfItemNotInView(menuDiv.current, "up");
       }
     });
 
-    function scrollUpIfTooHigh(
+    function scrollMenuIfItemNotInView(
       menuDivElement: HTMLDivElement,
-      itemDivElement: HTMLDivElement,
+      direction: string,
     ) {
-      const tempDiv = itemDivElement.parentElement as HTMLDivElement;
+      const tempDiv = menuDivElement.getElementsByClassName(
```

Comment:
I _think_ you may be better doing something like:

```ts
const option = menuDivElement.querySelector(`${CLASSNAME}:nth-child(${highlightedIndex})`)
```

This will push all the querying into the browser so it will be more efficient about it and we don't need an array here.

Reply:
Yup, that's a better way of query selecting; I'll do that!